src/contrib/cluster/Akka.Cluster.Sharding.Tests/PersistentShardSpec.cs

```diff
@@ -87,5 +136,96 @@ public void Persistent_Shard_must_remember_entities_started_with_StartEntity()
                 ExpectMsgAllOf(new Shard.ShardStats("shard-1", 1));
             });
         }
+
+        [Theory(DisplayName = "Persistent shard must recover from transient failures inside sharding entity constructor and PreStart method")]
+        [MemberData(nameof(PropsFactory))]
+        public async Task Persistent_Shard_must_recover_from_failing_entity(Props entityProp)
+        {
+            ExtractEntityId extractEntityId = message =>
+            {
+                switch (message)
+                {
+                    case ShardSpec.EntityEnvelope env:
+                        return (env.Id.ToString(), env.Payload);
+                }
+                return Option<(string, object)>.None;
+            };
+            
+            ExtractShardId extractShardId = message =>
+            {
+                switch (message)
+                {
+                    case ShardSpec.EntityEnvelope msg:
+                        return msg.Id.ToString();
+                }
+                return null;
+            };            
+            
+            var settings = ClusterShardingSettings.Create(Sys);
+            var tuning = settings.TuningParameters;
+            settings = settings.WithTuningParameters(new TuningParameters
```

Comment:
This is a very bad coding experience. We need to at least add a `Copy()` method to `TuningParameter` class so no one should be forced to write this. Or use the record feature in .NET 5.0